projects/components/src/datagrid/directives/select-directive.ts

```diff
@@ -0,0 +1,46 @@
+/*!
+ * Copyright 2019 VMware, Inc.
+ * SPDX-License-Identifier: BSD-2-Clause
+ */
+
+import { OnChanges } from '@angular/core';
+import { EventEmitter } from '@angular/core';
+import { Directive, Input } from '@angular/core';
+import { ClrDatagrid } from '@clr/angular';
+import { Host } from '@angular/core';
+import { GridSelectionType } from '../datagrid.component';
+
+@Directive({
+    selector: '[vcdComponentGridSelection]',
+})
+export class ComponentGridSelectionDirective<R> implements OnChanges {
+    @Input() gridSelectionType: GridSelectionType;
+
+    @Input() gridSelectionChanged: EventEmitter<R[]>;
+
+    /**
+     * The value of the single selection.
+     */
+    singleSelected: R;
+
+    /**
+     * The value of the multi selection.
+     */
+    multiSelection: R[] = [];
+
+    constructor(@Host() private datagrid: ClrDatagrid) {}
+
+    ngOnChanges(): void {
+        if (this.gridSelectionType === GridSelectionType.Single) {
```

Comment:
`ngOnChanges` isn't great, gets called when any inputs change? Consider using a setter instead. You should at least check the changes parameter in the cases where a setter doesn't work very well which I don't think is the case here.

Reply:
We're adding a subscription every time a bound property changes? We should also remove subscriptions when the component is destroyed. Consider trying always subscribing to both, and if needed, not do anything instead of adding and removing subscriptions. Let's chat if it's not clear.

See https://medium.com/@mendesjuan/a-better-way-to-unsubscribe-646b39c6b779